activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -188,6 +245,7 @@ def inherited(child_class)
           super
           child_class.initialize_generated_modules
           child_class.class_eval do
+            @alias_attributes_mass_generated = false
```

Comment:
Does this generate duplicated alias methods in child classes? (Do we care?)

Reply:
>Do we care?

Ultimately I believe we do. But since I believe it can't lead to an issue apart from doing a bit of unnecessary work I'd prefer to make improvements to this logic in a separate PR since I can imagine it's pretty rare for an application to use long inheritance chains of Active Record models along with alias attributes in it. At least I don't expect it to be noticeable. 

Overall we do need to reset the flag in child classes so alias attributes defined on a child class can be mass generated as well. As a fix we could most likely keep track of a list similar to `attribute_aliases` but this one will not keep track of parent-defined alias attributes and by using such list we will only be defining alias attribute methods that belong to the child. 
Or perhaps a check if the method has already been defined by parent will suffice.